st2actions/st2actions/workflows/workflows.py

```diff
@@ -78,13 +92,69 @@ def process(self, message):
             raise ValueError(msg)
 
         try:
+            with self._semaphore:
+                self._active_messages += 1
             handler_function(message)
         except Exception as e:
             # If the exception is caused by DB connection error, then the following
             # error handling routine will fail as well because it will try to update
             # the database and fail the workflow execution gracefully. In this case,
             # the garbage collector will find and cancel these workflow executions.
             self.fail_workflow_execution(message, e)
+        finally:
+            with self._semaphore:
+                self._active_messages -= 1
+
+    def start(self, wait):
+        spawn_after(self._delay, self._resume_workflows_paused_during_shutdown)
+        super(WorkflowExecutionHandler, self).start(wait=wait)
+
+    def shutdown(self):
+        super(WorkflowExecutionHandler, self).shutdown()
+        exit_timeout = cfg.CONF.workflow_engine.exit_still_active_check
+        sleep_delay = cfg.CONF.workflow_engine.still_active_check_interval
```

Comment:
Can we rename `still_active_check_interval` to `exit_check_interval`?  And rename `sleep_delay` to the same for consistency?

Reply:
We have same config option defined for action runner graceful shutdown as well that's why I reused the same name here.